### ISSUE TITLE
Tell the user to refresh the applet cache

### ISSUE THREAD

**Issue by mockturtl** [linuxmint/Cinnamon#2215]
When an applet update is available, clicking the update button gives a 404.

> "An error occurred while trying to access the server.  Please try again in a little while.
> Details:  HTTP Error 404: Not Found"

[Spices.py#download](https://github.com/linuxmint/Cinnamon/blob/747a2d81bc66a8a8c4b5edf1bd346a247f2fe8a2/files/usr/lib/cinnamon-settings/bin/Spices.py#L603)

I've had a few reports from users who don't understand they need to click "Refresh list" before updating.

Maybe the error message should be more informative?

**Comment by dalcde** [comment]
This also happens when you try to install applets without refreshing the cache list

**Comment by coryhilliard** [comment]
Dalcde is right.  You need to refresh list first.

**Comment by jaszhix** [comment]
I recently reproduced this issue, I am unable to update an applet I published recently. It does say an update is available, but refreshing the app, or restarting Cinnamon doesn't help. My guess is side-loading the applet from another source is causing it. Someone opened an [issue](https://github.com/jaszhix/icingtaskmanager/issues/2) on my applet's repo saying they were able to reproduce it.

I'm using Cinnamon 3.0.7 on Mint 18.
